sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageExtensions.cs

```diff
@@ -147,7 +94,7 @@ public static BinaryData GetBody(this AmqpAnnotatedMessage message)
         {
             if (message.Body.TryGetData(out IEnumerable<ReadOnlyMemory<byte>> dataBody))
             {
-                return dataBody.ConvertAndFlattenData();
+                return BinaryData.FromBytes((BodyMemory)dataBody);
```

Comment:
Of course this is a bit brute force right now especially since users can smuggle in stuff, but I figured it is good enough to get the conversation started even if it makes some test bomb with an InvalidCastException

Reply:
Yeah if users directly modify the AmqpAnnotatedMessage, this would not be BodyMemory.